src/esm_runscripts/filedicts.py

```diff
@@ -63,10 +67,68 @@ def cp(self) -> None:
     def ln(self) -> None:
         pass
 
-    def mv(self) -> None:
-        pass
-        
-        
+    def mv(self, source: str, target: str) -> None:
+        """
+        Moves (renames) the SimulationFile from it's location in ``source`` to
+        it's location in ``target``.
+
+        Parameters
+        ----------
+        source : str
+            One of ``"pool"``, ``"work"``, ``"exp_tree"``, "``run_tree``"
+        target : str
+            One of ``"pool"``, ``"work"``, ``"exp_tree"``, "``run_tree``"
+        """
+        if source not in self.locations:
+            user_error(
+                "Filedict Error",
+                "source is incorrectly defined, and needs to be in {self.locations}",
+            )
```

Comment:
I would keep the ValueError because user_error lowers the testability of the function. user_error shoud be called from the driver routine instead. What about the builtin `FileNotFoundError`?

Reply:
I would say `ValueError`, as the value of the variable `source` is incorrect. That is fixed though now.